File: NCS-Course-2022/NcsCourse-Backend/Day22-Sep21-Hibernate-Jpa-Single-Table-mapping/src/main/java/com/hibernate/single/table/ContractEmployee.java
package com.hibernate.single.table;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue(value = "C")
public class ContractEmployee extends Employee{
	
	@Column
	private int hourlyRate;

	public ContractEmployee() {
	}

	public ContractEmployee(int hourlyRate) {
		super();
		this.hourlyRate = hourlyRate;
	}

	public int getHourlyRate() {
		return hourlyRate;
	}

	public void setHourlyRate(int hourlyRate) {
		this.hourlyRate = hourlyRate;
	}

	@Override
	public String toString() {
		return "ContractEmployee [hourlyRate=" + hourlyRate + "]";
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = super.hashCode();
		result = prime * result + Objects.hash(hourlyRate);
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (!super.equals(obj))
			return false;
		if (getClass() != obj.getClass())
			return false;
		ContractEmployee other = (ContractEmployee) obj;
		return hourlyRate == other.hourlyRate;
	}

	
}

File: NCS-Course-2022/NcsCourse-Frontend/Day29-Oct3/contract-employee.ts
import Employee1 from "./Employee";

class contractEmployee extends Employee1 {
  private hourlyRate: number;

  constructor(id: number, name: string, hourlyRate: number) {
    super(id, name);
    this.hourlyRate = hourlyRate;
  }

  public showEmp(): void {
    super.showEmp();
    console.log("hourlyRate = " + this.hourlyRate);
  }
}

const obj: contractEmployee = new contractEmployee(101, "string", 900);
obj.showEmp();
